src/plugins/intel_cpu/src/transformations/cpu_opset/arm/pass/grid_sample_decomposition.cpp

```diff
@@ -0,0 +1,813 @@
+    return std::make_shared<op::v1::Select>(size_is_one, ctx.c0, res);
+}
+
+// Clamp integer-ish indices to [0..size-1] (safe)
+void clamp_indices_inplace(const Ctx& ctx,
+                           std::shared_ptr<Node>& xi,
+                           std::shared_ptr<Node>& yi,
+                           const std::shared_ptr<Node>& w_in_f,
+                           const std::shared_ptr<Node>& h_in_f) {
+    auto w_m1 = std::make_shared<op::v1::Subtract>(w_in_f, ctx.c1);
+    auto h_m1 = std::make_shared<op::v1::Subtract>(h_in_f, ctx.c1);
+    xi = std::make_shared<op::v1::Minimum>(std::make_shared<op::v1::Maximum>(xi, ctx.c0), w_m1);
+    yi = std::make_shared<op::v1::Minimum>(std::make_shared<op::v1::Maximum>(yi, ctx.c0), h_m1);
+}
+
+// Inside mask for ZEROS padding (check BEFORE clamp)
+std::shared_ptr<Node> inside_mask_indexed(const Ctx& ctx,
+                                          const std::shared_ptr<Node>& xi,
+                                          const std::shared_ptr<Node>& yi,
+                                          const std::shared_ptr<Node>& w_in_f,
+                                          const std::shared_ptr<Node>& h_in_f) {
+    auto x_ge_0 = std::make_shared<op::v1::GreaterEqual>(xi, ctx.c0);
+    auto y_ge_0 = std::make_shared<op::v1::GreaterEqual>(yi, ctx.c0);
+    auto x_lt_w = std::make_shared<op::v1::Less>(xi, w_in_f);
+    auto y_lt_h = std::make_shared<op::v1::Less>(yi, h_in_f);
+    return std::make_shared<op::v1::LogicalAnd>(std::make_shared<op::v1::LogicalAnd>(x_ge_0, x_lt_w),
+                                                std::make_shared<op::v1::LogicalAnd>(y_ge_0, y_lt_h));
+}
+
+// ---- builders for interpolation modes ----
+
+// NEAREST (HALF_TO_EVEN rounding; reflect after rounding; always clamp)
+std::shared_ptr<Node> build_nearest_nhwc(const Ctx& ctx, const ov::op::v9::GridSample::Attributes& attrs) {
+    const bool is_zeros = attrs.padding_mode == op::v9::GridSample::PaddingMode::ZEROS;
+    const bool is_reflection = attrs.padding_mode == op::v9::GridSample::PaddingMode::REFLECTION;
+
+    auto w_in_f = to_f32(ctx.w_in);
+    auto h_in_f = to_f32(ctx.h_in);
+
+    std::shared_ptr<Node> x_idx = std::make_shared<op::v5::Round>(ctx.x, op::v5::Round::RoundMode::HALF_TO_EVEN);
+    std::shared_ptr<Node> y_idx = std::make_shared<op::v5::Round>(ctx.y, op::v5::Round::RoundMode::HALF_TO_EVEN);
+
+    std::shared_ptr<Node> x_unclamped = x_idx;
+    std::shared_ptr<Node> y_unclamped = y_idx;
+
+    if (is_reflection) {
+        x_idx = reflect_index(ctx, x_idx, w_in_f, attrs.align_corners);
+        y_idx = reflect_index(ctx, y_idx, h_in_f, attrs.align_corners);
+    }
+    clamp_indices_inplace(ctx, x_idx, y_idx, w_in_f, h_in_f);
+
+    std::shared_ptr<Node> out = gather_hw_nhwc(ctx, x_idx, y_idx);
+
+    if (is_zeros) {
+        auto w_m1 = std::make_shared<op::v1::Subtract>(w_in_f, ctx.c1);
+        auto h_m1 = std::make_shared<op::v1::Subtract>(h_in_f, ctx.c1);
```

Comment:
In `inside_mask_indexed` the upper bound is `size`, here `size - ctx.c1` is used. Is it a misalignment?